src/main/java/ch/njol/skript/expressions/ExprDurability.java

```diff
@@ -36,78 +36,80 @@
 
+	private boolean durability;
+
 	static {
-		register(ExprDurability.class, Long.class, "((data|damage)[s] [value[s]]|durabilit(y|ies))", "itemtypes/blocks/slots");
+		register(ExprDurability.class, Long.class, "(damage[s] [value[s]]|durability:durabilit(y|ies))", "itemtypes/slots");
 	}
-	
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		durability = parseResult.hasTag("durability");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
 	@Override
 	@Nullable
-	public Long convert(final Object o) {
-		if (o instanceof Slot) {
-			final ItemStack i = ((Slot) o).getItem();
-			return i == null ? null : (long) ItemUtils.getDamage(i);
-		} else if (o instanceof ItemType) {
-			ItemStack item = ((ItemType) o).getRandom();
-			return (long) ItemUtils.getDamage(item);
+	public Long convert(Object object) {
+		ItemStack itemStack = null;
+		if (object instanceof Slot) {
+			itemStack = ((Slot) object).getItem();
+		} else if (object instanceof ItemType) {
+			itemStack = ((ItemType) object).getRandom();
 		}
-		return null;
+		if (itemStack == null)
+			return null;
+		long damage = ItemUtils.getDamage(itemStack);
+		return durability ? itemStack.getType().getMaxDurability() - damage : damage;
 	}
-	
+
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		switch (mode) {
-			case ADD:
-			case SET:
-			case RESET:
-			case REMOVE:
-			case DELETE:
-				return CollectionUtils.array(Number.class);
-		}
-		return null;
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.REMOVE_ALL)
+			return null;
+		return CollectionUtils.array(Number.class);
 	}
-	
-	@SuppressWarnings("null")
+
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		int a = delta == null ? 0 : ((Number) delta[0]).intValue();
-		final Object[] os = getExpr().getArray(e);
-		for (final Object o : os) {
+	@SuppressWarnings("null")
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		int i = delta == null ? 0 : ((Number) delta[0]).intValue();
+		Object[] objects = getExpr().getArray(event);
+		for (Object object : objects) {
 			ItemStack itemStack = null;
-			Block block = null;
-			
-			if (o instanceof ItemType)
-				itemStack = ((ItemType) o).getRandom();
-			else if (o instanceof Slot)
-				itemStack = ((Slot) o).getItem();
-			else
+
+			if (object instanceof ItemType) {
+				itemStack = ((ItemType) object).getRandom();
+			} else if (object instanceof Slot) {
+				itemStack = ((Slot) object).getItem();
+			}
+			if (itemStack == null)
 				return;
-			
-			int changeValue = itemStack != null ? ItemUtils.getDamage(itemStack) : block != null ? block.getData() : 0;
-			
+
+			int changeValue = ItemUtils.getDamage(itemStack);
+			if (durability)
+				changeValue -= itemStack.getType().getMaxDurability();
```

Comment:
Shouldn't this be the other way around, max durability - damage?

Reply:
oh yeah, it used to be like that but i must've changed it by accident, oopsie